main.tf

```diff
@@ -88,6 +88,10 @@ module "configure_user" {
 
 locals {
   is_storage_delegated = var.enable_cos_kms_encryption ? true : false
+  # fetch KMS region from existing_cos_kms_key_crn
+  kms_region = var.cos_kms_key_crn != null ? module.cos_kms_key_crn_parser[0].region : null
+  # tflint-ignore: terraform_unused_declarations
+  validate_kms_key_and_watsonx_ai_region = var.cos_kms_key_crn != null ? (local.kms_region == var.region ? true : tobool("KMS instance need to be in the same region as of watsonx.ai")) : true
```

Comment:
Can we handle this validation in the `region` variable and avoid `tobool(...)` as we are using `required_version = ">= 1.9.0"` ? 

Once done, comment to disable linting can be removed too. i.e. `# tflint-ignore: terraform_unused_declarations`

Reply:
done